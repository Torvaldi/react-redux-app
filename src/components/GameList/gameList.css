
/** General block **/
.block_gamelist, .block_gamelist_disable {
    list-style: none;
    display: flex;
    justify-content: space-around;
    margin: 20px 0;
    padding: 20px;
    box-shadow: 2px 2px 5px 0px rgba(0,0,0,0.75);
    animation-name: fadeIn;
    animation-duration: 800ms;
}

/** Default block **/
.block_gamelist {
    background-color:var(--darkBlue);
}

/** Disable block **/
.block_gamelist_disable {
    background-color: rgba(90, 90, 90, 0.75);
}

.block_gamelist_disable .config_detail_data {
    color: grey;
}

.block_gamelist_disable .gamelist_title {
    color: grey;
}

.gamelist_info_block {
    display: flex;
    align-items: baseline;
    margin-bottom: 5px;
}

.gamelist_title{
    margin: 3px 0;
    color: white;
}

.gamelist_time {
    color: var(--darkGrey);
}

.time_detail {
    font-weight: bold;
}

/* section inside gamelist */

.gamelist_join {
    width: 10%;
    display: flex;
    align-items: center;
    justify-content: center;
}

.gamelist_info {
    width: 35%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 0px 15px;
}

.gamelist_config {
    display: flex;
    width: 55%;
}

.config_detail {
    width: 33%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.config_detail_title {
    color: var(--darkGrey);
    font-size: 1rem;
    display: flex;
    align-items: center;
}

.config_detail_title_text {
    margin-left: 3px;
}

.config_detail_data {
    font-weight: bold;
    font-size: 1.6rem;
    padding: 10px;  
    color: white;
}

.status_title {
    color: var(--secondary);
    margin-left: 10px;
}

/* mobile */
@media only screen and (max-width: 550px) {
    .block_gamelist, .block_gamelist_disable, .gamelist_config {
      flex-direction: column;
      align-items: center;
      
    }

    .gamelist_info{
        text-align: center;
        width: 100%;
        margin-bottom: 5px;
    }



}