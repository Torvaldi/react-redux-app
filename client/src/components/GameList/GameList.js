import React from 'react';
import propTypes from 'prop-types';

import { getGamelistDateText } from '../../helper/date';
import { getGameStatus } from '../../helper/game';

import './gameList.css'
import Button from '@material-ui/core/Button';

import GroupIcon from '@material-ui/icons/Group';
import TrendingUpIcon from '@material-ui/icons/TrendingUp';
import ViewModuleIcon from '@material-ui/icons/ViewModule';
import StarIcon from '@material-ui/icons/Star';

import InfoBlock from './InfoBlock/InfoBlock';

const GameList = ({
  joinGame, 
  gameType, 
  game: { id, creator, created_at, level, answer, score_to_win, total_player, status}
  }) => {
      // get date under the form of "created at x minutes ago"
      let textCreationDate = getGamelistDateText(created_at);
      
      // dynamic style
      var block_gamelist = 'block_gamelist';
      var disabled = false;
      if(gameType === 'disable'){
        block_gamelist = 'block_gamelist_disable';
        disabled = true;
      }

      if(gameType === 'running'){
        var TextStatus = <span className="status_title">{getGameStatus(status)}</span>
      }
      
      return(
          <li className={block_gamelist} key={id} >
            <section className="gamelist_info">
              <div className="gamelist_info_block">
                <h2 className="gamelist_title">{creator}'s game</h2>
                {TextStatus ? TextStatus : ''}
              </div>
              <span className="gamelist_time"> 
                {textCreationDate}
              </span>
            </section>
            <section className="gamelist_config">
              <InfoBlock 
                icon={<GroupIcon />}
                title={"Players"}
                data={total_player + '/10'}
              />
              <InfoBlock 
                icon={<TrendingUpIcon />}
                title={"Level"}
                data={level}
              />
              <InfoBlock 
                icon={<ViewModuleIcon />}
                title={"Answers"}
                data={answer}
              />
              <InfoBlock 
                icon={<StarIcon />}
                title={"Winning Score"}
                data={score_to_win}
              />
            </section>
            <section className="gamelist_join">
              <form onSubmit={joinGame(id)} >
              <Button 
              type="submit" 
              size="medium" 
              variant="contained" 
              color="secondary"
              disabled={disabled}
              >
                Join
              </Button>
              </form>
            </section>
          </li>
      );
    
}

GameList.propTypes = {
  joinGame: propTypes.func.isRequired,
  gameType: propTypes.string.isRequired,
  game: propTypes.object.isRequired,
}

export default GameList
