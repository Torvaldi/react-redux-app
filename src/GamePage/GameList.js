import React from 'react';
import './css/gameList.css'
import { getDifferenceCurrentDate } from '../helper/date';

const GameList = ({game : {id, userCreator, timestamp, level, answer, scoreToWin} }) => {

    let date = getDifferenceCurrentDate(timestamp);

    return (
      <li className="block_gamelist" key={id}>
          <section className="gamelist_info">
            <h2>{userCreator.username} 's game</h2>
            <span className="gamelist_time">Created 
              <span className="time_detail"> {date.minutes} </span> 
              minutes and 
              <span className="time_detail"> {date.seconds} </span>seconds ago
            </span>
          </section>
          <section className="gamelist_config">

          </section>
      </li>
    );
}

export default GameList;
