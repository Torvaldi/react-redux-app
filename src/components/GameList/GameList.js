import React from 'react';
import { getGamelistDateText } from '../../helper/date';
import { getGameStatus } from '../../helper/game';

import './gameList.css'
import Button from '@material-ui/core/Button';

const GameList = ({
  joinGame, 
  gameType, 
  game: { id, creator, timestamp, level, answer, score_to_win, total_player, status}
  }) => {
      // get date under the form of "created at x minutes ago"
      let textCreationDate = getGamelistDateText(timestamp);
      
      // dynamic style
      var block_gamelist = 'block_gamelist';
      var disabled = false;
      if(gameType == 'disable'){
        block_gamelist = 'block_gamelist_disable';
        disabled = true;
      }

      if(gameType == 'running'){
        var TextStatus = <span className="status_title"> - {getGameStatus(status)}</span>
      }

      return(
        <li className={block_gamelist} key={id}>
            <section className="gamelist_info">
              <h2 className="gamelist_title">
              {creator} 's game
              {TextStatus ? TextStatus : ''}
              </h2>
              <span className="gamelist_time"> 
                {textCreationDate}
              </span>
            </section>
            <section className="gamelist_config">
              <article className="config_detail">
                <span className="config_detail_title">Players</span>
                <span className="config_detail_data" >{total_player}/10</span>
              </article>
              <article className="config_detail">
                <span className="config_detail_title">Level</span>
                <span className="config_detail_data" >{level}</span>
              </article>
              <article className="config_detail">
                <span className="config_detail_title">Answers</span>
                <span className="config_detail_data">{answer}</span>
              </article>
              <article className="config_detail">
                <span className="config_detail_title">Winning Score</span>
                <span className="config_detail_data">{score_to_win}</span>
              </article>
            </section>
            <section className="gamelist_join">
              <form onSubmit={joinGame(id)} >
              <Button 
              type="submit" 
              size="medium" 
              variant="contained" 
              color="secondary"
              disabled={disabled}
              >
                Join
              </Button>
              </form>
            </section>
        </li>
      );
    
}


export default GameList
