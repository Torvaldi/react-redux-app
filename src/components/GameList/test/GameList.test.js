import React, { Fragment } from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import GameList from '../GameList';

configure({ adapter: new Adapter() });

describe('<GameList />', () => {

    // snapshot & props testing
    it('should render GameList Componenent', () => {
        let joinGame = () => {};
        let gameType = 'disable';
        let game = { id: 1, 
            creator: "Bob", 
            timestamp: "2019-07-22 13:52:36", 
            level: 2, 
            answer: 12, 
            score_to_win: 100, 
            total_player: 4, 
            status: 1 
        }
        const wrapper = shallow(<GameList joinGame={joinGame} gameType={gameType} game={game} />);
        expect(wrapper).toMatchSnapshot();

        const gameListComponent = mount(<GameList joinGame={joinGame} gameType={gameType} game={game} />);
        expect(gameListComponent.prop('joinGame')).toEqual(joinGame); 
        expect(gameListComponent.prop('gameType')).toEqual(gameType); 
        expect(gameListComponent.prop('game')).toEqual(game); 
    });
    
});