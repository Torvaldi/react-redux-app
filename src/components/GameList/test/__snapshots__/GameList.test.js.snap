// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameList /> should render GameList Componenent 1`] = `ShallowWrapper {}`;
